client/services/Hooks.ts

```diff
@@ -99,11 +99,27 @@ function withAuthModal(isLoggedIn: boolean, cb: (...any) => any) {
   };
 }
 
+function pressEsc(ref: MutableRefObject<HTMLElement>, handler: (e: MouseEvent) => void) {
```

Comment:
Couple of problems here:

1. Why do we need the `ref` parameter? It's not used at all.
2. The type `e: MouseEvent` is wrong. It should be a keyboard event.

And finally, let's rename to `useKeyDown` and make it generic, so it can accept any key. This way we can reuse it in different scenarios.